<launch>
    <include file="$(find turtlebot_bringup)/launch/minimal.launch"/>
    <!--include file="$(find turtlebot_bringup)/launch/3dsensor.launch"/-->
    <include file="$(find turtlebot_navigation)/launch/amcl_demo.launch">
        <arg name="map_file" value = "/home/humphrey/Documents/catkin_ws/src/c3/map.yaml"/>
    </include>
    <include file="$(find turtlebot_rviz_launchers)/launch/view_navigation.launch"/>
    
    <arg name="marker_size" default="19.9" />
    <arg name="max_new_marker_error" default="0.08" />
    <arg name="max_track_error" default="0.2" />
    <arg name="cam_image_topic" default="/camera/rgb/image_raw" />
    <arg name="cam_info_topic" default="/camera/rgb/camera_info" />
    <arg name="output_frame" default="/base_link" />

    <node name="work4" pkg="c3" type="work4.py" output="screen"/>

    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
        <param name="marker_size"           type="double" value="$(arg marker_size)" />
        <param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
        <param name="max_track_error"       type="double" value="$(arg max_track_error)" />
        <param name="output_frame"          type="string" value="$(arg output_frame)" />

        <remap from="camera_image"  to="$(arg cam_image_topic)" />
        <remap from="camera_info"   to="$(arg cam_info_topic)" /> 
    </node>
    
</launch>